pytest_trio/plugin.py

```diff
@@ -307,8 +315,69 @@ async def run(self, test_ctx, contextvars_ctx):
                     raise RuntimeError("too many yields in fixture")
 
 
+def _trio_test(run):
+    """Use:
+        @trio_test
+        async def test_whatever():
+            await ...
+
+    Also: if a pytest fixture is passed in that subclasses the ``Clock`` abc, then
+    that clock is passed to ``trio.run()``.
+    """
+
+    def decorator(fn):
+        @wraps(fn)
+        def wrapper(**kwargs):
+            __tracebackhide__ = True
+            clocks = [c for c in kwargs.values() if isinstance(c, Clock)]
+            if not clocks:
+                clock = None
+            elif len(clocks) == 1:
+                clock = clocks[0]
+            else:
+                raise ValueError("too many clocks spoil the broth!")
+            instruments = [
+                i for i in kwargs.values() if isinstance(i, Instrument)
+            ]
+            return run(
+                partial(fn, **kwargs), clock=clock, instruments=instruments
+            )
+
+        return wrapper
+
+    return decorator
+
+
 def _trio_test_runner_factory(item, testfunc=None):
-    testfunc = testfunc or item.obj
+    if testfunc:
+        run = trio.run
+    else:
+        testfunc = item.obj
+
+        config_run = choose_run(config=item.config)
+        runs = {
+            marker.kwargs.get('run', config_run)
+            for marker in item.iter_markers("trio")
+        }
+
+        if len(runs) == 0:  # pragma: no cover
+            raise RuntimeError(
+                "No 'trio' marker found.  Please report to pytest-trio."
+            )
+        elif len(runs) == 1:
+            [run] = runs
+        else:
+            runs.discard(trio.run)
+            if len(runs) == 1:
+                [run] = runs
+            else:
+                runs_string = ', '.join(
+                    sorted(f"{f.__module__}.{f.__qualname__}" for f in runs)
+                )
+                raise ValueError(
+                    "Not yet able to select from more than one third-party" +
```

Comment:
You can concatenate strings by just putting them next to each other, no `+` needed. (I generally prefer to f-prefix all strings in such combinations if any of them use interpolation, because otherwise it's too easy to reflow one of the {} components to a line with a different prefix when adding or removing text.)